add accounts mvc

DeryckSPBU opened the pull request:
Working with accounts mvc and jsp pages for accounts

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/controller/AccountController.java:
```diff
@@ -0,0 +1,47 @@
+  private static final String ACCOUNT_PAGE = "account";
+  private static final String ADD_ACCOUNT_PAGE = "addAccount";
+
+  private AccountService accountService;
```
I think it's missing `@Autowired`

nikandtos replied:
why?

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/service/impl/AccountServiceImpl.java:
```diff
@@ -0,0 +1,50 @@
+@Service
+public class AccountServiceImpl implements AccountService {
+
+  private AccountDao accountDao;
```
I think it's missing `@Autowired`

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/controller/AccountController.java:
```diff
@@ -0,0 +1,47 @@
+  @PostMapping(value = "/addAccount")
+  public String addAccount(@RequestParam(name = "amount") double accountAmount) {
+
+    Account account = new Account();
```
You can use only `accountService.createAccount(amount)'

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/service/AccountService.java:
```diff
@@ -0,0 +1,11 @@
+public interface AccountService {
+  Account createAccount(Account account);
+
+  List<Account> getAllAccounts();
```
Why do we need this operation? I think we should do it by user

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/service/impl/AccountServiceImpl.java:
```diff
@@ -0,0 +1,50 @@
+   * @param isActive boolean flag
+   * @return List of accounts
+   */
+  public List<Account> getAllAccounts(boolean isActive) {
```
I think, we should do it by user and also it should be declared in interface

DeryckSPBU commented:
Added user and account pages with list of theirs linked  entities.
Added account creation, unit tests for account service layer,
Also added logger, i18n, javadocs, html views.

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/service/AccountService.java:
```diff
@@ -1,11 +1,19 @@
+
+  Account getAccountById(long id);
+
+  List<Card> getAllCardsByAccount(Account account);
```
This is not supposed to be here

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/service/CardService.java:
```diff
@@ -0,0 +1,16 @@
+
+  Card createCard(Card card);
+
+  Account getAccountByCard(Card card);
```
This is not supposed to be here

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/service/impl/AccountServiceImpl.java:
```diff
@@ -1,50 +1,105 @@
     accountToCreate.setActive(true);
-    return accountToCreate;
-  }
+    LOGGER.info("Account has benn created");
```
![451](https://user-images.githubusercontent.com/43749789/47990609-ff079b80-e0f8-11e8-9b29-02c75c6eadca.jpg)

DeryckSPBU commented on src/main/java/com/epam/lab/paymentsystem/service/impl/AccountServiceImpl.java:
```diff
@@ -0,0 +1,97 @@
+   * @throws UnsupportedOperationException if there's not enough money on source account
+   */
+  @Override
+  @Transactional
```
To provide transaction for money transfer

DeryckSPBU commented on src/main/java/com/epam/lab/paymentsystem/service/impl/AccountServiceImpl.java:
```diff
@@ -0,0 +1,97 @@
+      LOGGER.error("Passed negative amount of account");
+      throw new UnsupportedOperationException("Amount should be positive");
+    }
+    accountToCreate.setAmount(accountDto.getAmount());
```
Absolutely.

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/service/impl/AccountServiceImpl.java:
```diff
@@ -65,21 +59,34 @@ public Account createAccount(AccountDto accountDto) throws UnsupportedOperationE
    */
   @Override
-  @Transactional
   public void makeTransaction(Account source, Account target, long amount)
```
https://stackoverflow.com/questions/30868829/is-it-possible-to-invoke-transactional-method-from-non-transactional-method-in-s

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/service/impl/AccountServiceImpl.java:
```diff
@@ -0,0 +1,97 @@
+  public void makeTransaction(Account source, Account target, long amount)
+      throws UnsupportedOperationException {
+    LOGGER.info("Creating transaction");
+    if (source.getAmount() - amount < 0) {
```
You have to move "check-logic" to `makeTransaction` method too
Also could you add more brackets so it would look this way?
`if ((source.getAmount() - amount) < 0)`
